_studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp

```diff
@@ -833,11 +833,14 @@ namespace
 #if defined(MFX_ENABLE_MFE)
     mfxU16 GetDefaultNumMfeFrames(mfxU32 targetUsage, const mfxFrameInfo& info,
         eMFXHWType platform, mfxFeiFunction func, int slices, bool extSurfUsed,
-        eMFXGTConfig config)
+        eMFXGTConfig config, mfxU16 brcMethod)
     {
         targetUsage;//no specific check for TU now, can be added later
         if (platform <= MFX_HW_BDW)//no MFE support prior to SKL.
             return 1;
+        // ICQ is unsupported for MFE now, QVBR and VCM are low latency usage rate controls, MFE is not suitable for low latency usages
+        if(brcMethod == MFX_RATECONTROL_ICQ || brcMethod == MFX_RATECONTROL_QVBR || brcMethod == MFX_RATECONTROL_VCM)
+            return 1;
```

Comment:
I am confused now. If you see https://github.com/Intel-Media-SDK/MediaSDK/issues/863 , it is just 1:1 transcoding case. MFE should not take effect. So I guess anywhere else should also be check: if only one channel encoding, don't entry MFE path.

Reply:
ffmpeg enables mfe path by default, while enabling ICQ, it wasn't validated for interoperability with MFE, but in result ffmpeg crashing.

Reply:
FFmpeg is just set MFX_MF_AUTO with the exception to be disabled in same cases. MSDK can be free to disable it for such as case. Is there any benefit to go to MFE path for one channel encoding case?  Or it is hard to be detected ?